test-data/unit/check-functions.test

```diff
@@ -363,7 +363,7 @@ y = x  # E: Incompatible types in assignment (expression has type Callable[...,
 
 a, b = None, None # type: (A, B)
 a = f()     # E: Incompatible types in assignment (expression has type "B", variable has type "A")
-b = f(b)    # E: Argument 1 to "f" has incompatible type "B"; expected "A"
+b = f(b)    # E: Argument 1 to "f" has incompatible type "B"; expected "Optional[A]"
```

Comment:
I wonder if it makes more sense to change the definition of f() to use A() as the default value? Ditto for the next test below.